app/screens/receive-bitcoin-screen/receive-screen.tsx

```diff
@@ -0,0 +1,302 @@
+                    onPress={request.share}
+                  >
+                    <Text {...testProps("Share Invoice")} color={colors.grey2}>
+                      <Icon color={colors.grey2} name="share-outline" />
+                      <Text> </Text>
+                      {LL.ReceiveScreen.shareInvoice()}
+                    </Text>
+                  </TouchableOpacity>
+                </View>
+              </>
+            )}
+        </View>
+
+        <View style={styles.extraDetails}>
+          <Text>{request.extraDetails}</Text>
+        </View>
+
+        <ButtonGroup
+          selectedId={request.type}
+          buttons={[
+            {
+              id: Invoice.Lightning,
+              text: LL.ReceiveScreen.lightning(),
+              icon: "md-flash",
+            },
+            { id: Invoice.PayCode, text: LL.ReceiveScreen.paycode(), icon: "md-at" },
+            {
+              id: Invoice.OnChain,
+              text: LL.ReceiveScreen.onchain(),
+              icon: "logo-bitcoin",
+            },
+          ]}
+          onPress={(id) => request.setType(id as InvoiceType)}
+          style={styles.invoiceTypePicker}
+        />
+        <AmountInput
+          unitOfAccountAmount={request.unitOfAccountAmount}
+          setAmount={request.setAmount}
+          canSetAmount={request.canSetAmount}
+          convertMoneyAmount={request.convertMoneyAmount}
+          walletCurrency={request.receivingWalletDescriptor.currency}
+          showValuesIfDisabled={false}
+          big={false}
+        />
+        <NoteInput
+          onBlur={request.setMemo}
+          onChangeText={request.setMemoChangeText}
+          value={request.memoChangeText || ""}
+          editable={request.canSetMemo}
+          style={styles.note}
+          big={false}
+        />
+
+        {OnChainCharge}
+
+        <SetLightningAddressModal
+          isVisible={request.isSetLightningAddressModalVisible}
+          toggleModal={request.toggleIsSetLightningAddressModalVisible}
+        />
+      </Screen>
+    </>
+  )
+}
+
+const useStyles = makeStyles(({ colors }) => ({
+  screenStyle: {
+    paddingHorizontal: 16,
+    paddingVertical: 12,
+    flexGrow: 1,
+  },
+  tabRow: {
+    flexDirection: "row",
+    flexWrap: "nowrap",
+    justifyContent: "center",
+    marginTop: 12,
+  },
+  usdActive: {
+    backgroundColor: colors.green,
+    borderRadius: 7,
+    justifyContent: "center",
+    alignItems: "center",
+    width: 150,
+    height: 30,
+    margin: 5,
+  },
+  btcActive: {
+    backgroundColor: colors.primary,
+    borderRadius: 7,
+    justifyContent: "center",
+    alignItems: "center",
+    width: 150,
+    height: 30,
+    margin: 5,
+  },
+  inactiveTab: {
+    backgroundColor: colors.grey3,
+    borderRadius: 7,
+    justifyContent: "center",
+    alignItems: "center",
+    width: 150,
+    height: 30,
+    margin: 5,
+  },
+  qrView: {
+    marginBottom: 10,
+  },
+  receivingWalletPicker: {
+    marginBottom: 10,
```

Comment:
For future reference, instead of spacing everything with margin, you can set the `rowGap` property on the top level container.